pkgs/servers/jetbrains/youtrack/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib, stdenv, fetchurl, fetchzip, makeWrapper, jdk17, p7zip, gawk, baseStatePath ? "/var/lib/youtrack" }:
+let
+  meta = with lib; {
+    description = "Issue tracking and project management tool for developers";
+    maintainers = teams.serokell.members ++ [ lib.maintainers.leona ];
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    # https://www.jetbrains.com/youtrack/buy/license.html
+    license = licenses.unfree;
+  };
+
+  generic = {
+    version, hash,
+  }: stdenv.mkDerivation rec {
+    pname = "youtrack";
+    inherit version;
+
+    # This base version is seperated by _ instaed of . to ensure path compatability and confusion
+    baseVersion = with lib; pipe version [
+      versions.splitVersion
+      (take 2)
+      (concatStringsSep "_")
+    ];
+
+    src = fetchzip {
+      url = "https://download.jetbrains.com/charisma/youtrack-${version}.zip";
+      inherit hash;
+    };
+
+    nativeBuildInputs = [ makeWrapper p7zip ];
```

Comment:
I haven't tried that myself but wouldn't `makeBinaryWrapper` work as well?

Reply:
Turns out: You don't even need `p7zip` because you already unpacked the zip in the fetcher

Reply:
Oh yes! `p7z`ip is actually a relict of a really old derivation i tried.